active/000-buisness-attributes/000-business-attributes.md

```diff
@@ -0,0 +1,258 @@
+                        "urn:li:tag:TestTag",
+                        "urn:li:tag:TestTag",
+                        "urn:li:tag:TestTag"
+                    ],
+                    "businessAttributeGlossary": [
+                        "urn:li:glossaryTerm:5fb9fd4e-5ed2-46e9-afa2-6ab5bb1f4851",
+                        "urn:li:glossaryTerm:2b441fe5-c466-479a-8ce7-5b3f2db7ce1a"
+                    ],
+                    "businessAttributeTags": [
+                        "urn:li:tag:TestTag",
+                        "urn:li:tag:TestTag"
+                    ],
+                    "businessAttributeDescription": [
+                        " quantiy-loremipsume",
+                        " productLoremipsume"
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+#### Cascade changes made in Business Attribute to related Datasets
+
+When any changes made to referenced entity, in this case Business attribute, for example, removal/add of tag/glossary-terms or description update, these changes needs to be cascaded to attached dataset elastic index. For this we are proposing to create either new annotation or using "`@SearchableRef`" with incoming and outgoing references like "`@Relationship`" entity, but under the hood we are planning to use "`@Relationship`" to fetch attached dataset entities and update their corresponding elastic indexes as per the changes made in referenced entity.
+
+
+## How we teach this
+
+For technical users, the concept of business attribute is best understood as user-defined abstract schema fields. In this context, anyone in the technology space can visualize it as the abstract record which can be inherited from by the implementation record, which is the actual physical schema field.
+
+For architects, these are best understood as field templates. They define a template for a field by assigning metadata to each field which corresponds to a specification.
+
+They are meant to be entirely transparent to business users and as such should not need to be introduced to them.
+
+While the change doesn't fundamentally alter any existing Datahub concept, it greatly empowers users and acts as an accelerator to enrichment. As such, it should be taught as a fundamentally new concept which can be leveraged by large Datahub customers to greatly reduce their metadata management overhead.
+
+## Drawbacks
+
+Cascading changes made in Business Attribute to related Dataset elastic indexes put a load on Kafka and elastic. Changes made in business attribute results in Kafka events, which gets consumes by the consumer which in turn update the elastic indexes of the referenced dataset with the updated changes of business attribute.  As per current implementation, Kafka consumer finalise the message processing until it updates the elastic index, so for example, if business attribute is referenced in datasets(order of 100K), then processing large volumes of messages can lead to performance degradation.
```

Comment:
Agree that this write amplification might make this approach infeasible to implement. Curious if you've run any benchmarks to see how this scales if you have 1M entities attached to a business attribute, and you have a change in the description of the attribute.

Reply:
With this approach we will have eventual consistency. When there is change, for example in business attribute description, now this change should be propagated to the elastic indexes for all the referenced datasets. During this propagation search experience might be inconsistent but after some time eventual consistency will be achieved. As of now we have not run any benchmarking. After design approval we will take care of testing and benchmarking during implementation time.